src/Operator/Constructors.jl

```diff
@@ -553,10 +693,96 @@ function diffusionoperator(basis::AbstractBasis, mesh::Mesh)
     return diffusion
 end
 
+"""
+```julia
+advectionoperator(basis, mesh)
+```
+Convenience constructor for advection operator
+
+# Weak form:
+- ``\\int \\nabla v u``
+
+# Arguments:
+- `basis`: basis for all operator fields to use
+- `mesh`:  mesh for operator
+
+# Returns:
+- Advection operator with basis on mesh
+
+# Example:
+```jldoctest
+# advection operator
+mesh = Mesh2D(1.0, 1.0);
+mapping = haletrefethenstriptransformation(1.4);
+basis = TensorH1LagrangeBasis(3, 4, 1, mesh.dimension, collocatedquadrature = false, mapping = mapping)
+advection = LFAToolkit.advectionoperator(basis, mesh);
+
+# verify
+println(advection)
+
+# output
+
+finite element operator:
+2d mesh:
+    dx: 1.0
+    dy: 1.0
+
+2 inputs:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    interpolation
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    quadratureweights
+
+1 output:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    gradient
+```
+"""
+function advectionoperator(basis::AbstractBasis, mesh::Mesh)
+    # setup
+    U = 1.0
+    function advectionweakform(u::Array{Float64}, w::Array{Float64})
+        dv = U * u * w[1]
```

Comment:
I don't think this is correct for 2D advection. You should make the velocity into a vector, and I'd make it an optional argument to the function. Something like
```suggestion
        dv = wind * u * w[1]
```